daemon/graphdriver/overlay/overlay.go

```diff
@@ -263,6 +300,20 @@ func (d *Driver) Create(id, parent string, opts *graphdriver.CreateOpts) (retErr
 		}
 	}()
 
+	driver := &Driver{}
+	if opts != nil {
+		if err := d.parseStorageOpt(opts.StorageOpt, driver); err != nil {
+			return err
+		}
+	}
+
+	if driver.options.quota.Size > 0 && !strings.HasSuffix(dir, "-init") {
+		// enforce container disk project quota
+		if err := d.quotaCtl.SetQuota(dir, driver.options.quota); err != nil {
```

Comment:
In order to make this work cleanly, I think we'll need to break the create logic out into a separate function that takes an `overlayOptions` as one of the arguments. The `(d *Driver) Create()` function will just pass in an empty options (until a time that there is some opt it should pass), and `(d *Driver) CreateReadWrite()` can pass in an `ovlerlayOptions` with the qutoa set.